gnosis/eth/tests/oracles/test_cowswap.py

```diff
@@ -0,0 +1,85 @@
+from unittest import mock
+
+from django.test import TestCase
+
+from eth_account import Account
+from requests import Session
+
+from ... import EthereumClient
+from ...oracles import CannotGetPriceFromOracle, CowswapOracle
+from ..ethereum_test_case import EthereumTestCaseMixin
+from ..test_oracles import (
+    dai_token_mainnet_address,
+    usdc_token_mainnet_address,
+    usdt_token_mainnet_address,
+    weth_token_mainnet_address,
+)
+from ..utils import just_test_if_mainnet_node
+
+
+class TestCowswapOracle(EthereumTestCaseMixin, TestCase):
+    def test_get_price(self):
+        mainnet_node = just_test_if_mainnet_node()
```

Comment:
OOC: is this testing the real node (E2E)?

Reply:
That's right, we are doing E2E for all oracle providers